support-lambdas/stripe-intent/cfn.yaml

```diff
@@ -0,0 +1,95 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: Merges together the salesforce account referenced by a set of zuora accounts
+
+Parameters:
+  Stage:
+    Description: Stage name
+    Type: String
+    AllowedValues:
+      - PROD
+      - CODE
+    Default: CODE
+
+Mappings:
+  StageMap:
+    CODE:
+      DomainName: stripe-intent-code.support.guardianapis.com
+      StripeConfig: test
+    PROD:
+      DomainName: stripe-intent.support.guardianapis.com
+      StripeConfig: live
+  Constants:
+    Alarm:
+      Process: Follow the process in https://docs.google.com/document/d/1_3El3cly9d7u_jPgTcRjLxmdG2e919zCLvmcFCLOYAk/edit
+      Urgent: URGENT 9-5 -
+
+Conditions:
+  CreateProdMonitoring: !Equals [ !Ref Stage, PROD ]
+
+Resources:
+  LambdaFunctionOverHttps:
+    Type: AWS::Serverless::Function
+    Properties:
+      Description: Returns a stripe setup intent token so we can get authorisation of a recurring payment on the client side
+      FunctionName:
+        !Sub stripe-intent-${Stage}
+      Handler: com.gu.stripeIntent.Handler::handleRequest
+      CodeUri:
+        Bucket: support-workers-dist
+        Key: !Sub support/${Stage}/stripe-intent/stripe-intent.jar
+      MemorySize: 1536
+      Runtime: java8
+      Timeout: 300
+      Policies:
+        - Statement:
+          - Effect: Allow
+            Action:
+              - logs:CreateLogGroup
+              - logs:CreateLogStream
+              - logs:PutLogEvents
+              - lambda:InvokeFunction
+            Resource: !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/stripe-intent-${Stage}:log-stream:*"
+        - Statement:
+          - Effect: Allow
+            Action: s3:GetObject
+            Resource:
+              - !Sub
+                - arn:aws:s3:::gu-reader-revenue-private/stripe/${stripeConfig}-stripe-regular.private.conf
+                - stripeConfig:
+                    !FindInMap [ StageMap, !Ref Stage, StripeConfig ]
+      Events:
+        HttpPost:
+          Type: Api
+          Properties:
+            Path: '/stripe-intent'
+            Method: post
+
+  DomainName:
+    Type: "AWS::ApiGateway::DomainName"
+    Properties:
+      RegionalCertificateArn: # only for *.support.guardianapis.com
+        !Sub arn:aws:acm:${AWS::Region}:${AWS::AccountId}:certificate/b384a6a0-2f54-4874-b99b-96eeff96c009
+      DomainName: !FindInMap [ StageMap, !Ref Stage, DomainName ]
+      EndpointConfiguration:
+        Types:
+          - REGIONAL
+
+  BasePathMapping:
+    Type: "AWS::ApiGateway::BasePathMapping"
+    Properties:
+      RestApiId: !Ref ServerlessRestApi # auto generated by the Transform
+      DomainName: !Ref DomainName
+      Stage: !Sub Prod
+    DependsOn: ServerlessRestApiProdStage # auto generated by the Transform
```

Comment:
this is a hack based on the predicted names of the low level resources.  AWS are working on proper support for custom domain names, at which point we can feel free to switch over to that https://github.com/awslabs/serverless-application-model/issues/783